fuel-txpool/src/txpool.rs

```diff
@@ -0,0 +1,473 @@
+            for rem in rem.iter() {
+                self.by_hash
+                    .remove(&rem.id())
+                    .expect("Expect to hash of tx to be present");
+                self.by_gas_price.remove(rem);
+            }
+
+            Ok(rem)
+        }
+    }
+
+    /// Return all sorted transactions that are includable in next block.
+    pub fn sorted_includable(&self) -> Vec<ArcTx> {
+        self.by_gas_price
+            .sort
+            .iter()
+            .rev()
+            .map(|(_, tx)| tx.clone())
+            .collect()
+    }
+
+    // When block is updated we need to receive all spend outputs and remove them from txpool
+    pub fn block_update(&mut self /*spend_outputs: [Input], added_outputs: [AddedOutputs]*/) {}
+
+    pub fn remove(&mut self, tx: &ArcTx) -> Vec<ArcTx> {
+        self.remove_by_tx_id(&tx.id())
+    }
+
+    /// remove transaction from pool needed on user demand. Low priority
+    pub fn remove_by_tx_id(&mut self, tx_id: &TxId) -> Vec<ArcTx> {
+        if let Some(tx) = self.by_hash.get(tx_id) {
+            self.by_gas_price.remove(tx);
+            return self
+                .by_dependency
+                .recursively_remove_all_dependencies(&self.by_hash, tx.clone());
+        }
+        Vec::new()
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use crate::Error;
+    use core::str::FromStr;
+    use fuel_core_interfaces::db::helpers::*;
+    use std::cmp::Reverse;
+
+    #[tokio::test]
+    async fn simple_insertion() {
+        let config = Arc::new(Config::default());
+        let db = DummyDB::filled();
+
+        let tx1_hash =
+            TxId::from_str("0x0000000000000000000000000000000000000000000000000000000000000010")
+                .unwrap();
+        let tx1 = Arc::new(DummyDB::dummy_tx(tx1_hash));
+
+        let mut txpool = TxPool::new(config);
+        let out = txpool.insert(tx1, &db).await;
+        assert!(out.is_ok(), "Transaction should be OK, get err:{:?}", out);
+    }
+
+    #[tokio::test]
+    async fn simple_dependency_tx1_tx2() {
+        let config = Arc::new(Config::default());
+        let db = DummyDB::filled();
+
+        let tx1_hash =
+            TxId::from_str("0x0000000000000000000000000000000000000000000000000000000000000010")
+                .unwrap();
+
+        let tx2_hash =
+            TxId::from_str("0x0000000000000000000000000000000000000000000000000000000000000011")
+                .unwrap();
+        let tx1 = Arc::new(DummyDB::dummy_tx(tx1_hash));
+        let tx2 = Arc::new(DummyDB::dummy_tx(tx2_hash));
+
+        let mut txpool = TxPool::new(config);
+
+        let out = txpool.insert(tx1, &db).await;
+        assert!(out.is_ok(), "Tx1 should be OK, get err:{:?}", out);
+        let out = txpool.insert(tx2, &db).await;
+        assert!(out.is_ok(), "Tx2 dependent should be OK, get err:{:?}", out);
+    }
+
+    #[tokio::test]
+    async fn missing_dep_faulty_tx1_tx2() {
```

Comment:
We should do better with our test names to clarify which requirements are being tested and what the expected behavior is. Referencing indices into helper data (`tx1`, `tx2`, etc) isn't a great way to clarify intent, and is harder to refactor if the test data changes over time.

Something like "insert_with_missing_input_coin_tx_fails" would be a lot more clear.

Reply:
These are data specific tests, if data changes it is like rewriting new test. But name of this particualr test is not the greatest, renamed it to: "fails_to_insert_tx2_with_missing_utxo_dependency_on_faulty_tx1"

Reply:
Ultimately what I'm getting at is we should rely more on conveying the requirements being tested, than the particular data being used. This is an improvement though but some of the other tests could probably use similar improvements.

Reply:
What test do you have in mind? btw to speed up this, be free to push commit dirrectly to branch i wouldn't mind, it is esentially change about preferences of test naming.